eventstore/storethread.go

```diff
@@ -85,7 +85,14 @@ func (a *singleThreadAdapter) threadToStore(wg *sync.WaitGroup) {
 
 			event, err := threadcbor.EventFromRecord(ctx, a.api, rec.Value())
 			if err != nil {
-				log.Fatalf("error when getting event from record: %v", err) // ToDo: Buffer them and retry...
+				block, err := rec.Value().GetBlock(ctx, a.api)
+				if err != nil { // ToDo: Buffer them and retry...
+					log.Fatalf("error when getting block from record: %v", err)
+				}
+				event, err = threadcbor.EventFromNode(block)
+				if err != nil {
+					log.Fatalf("error when decoding block to event: %v", err)
+				}
```

Comment:
Bang. Here we had a problem that was uncovered now that I fetch all the unkown records when we get a far ahead record for a log.

Records that aren't local, seems that never have the `Event` cached. So I should do here all the path to get it. Getting the raw block for the record, then unmarshal it to an Event.

Here is the ** I mentioned before. A `Record` method to get the underlying event maybe be useful here?

Reply:
By non-local events I mean events generated by *other* peers. And yes, records received by grpc service are fully loaded. 

But here's the tricky part: we receive the *head* fully loaded. If while adding the head we discover that we are missing extra records up to our known head, those missing records will be fetched from the `DAGService`, and *those* records aren't fully loaded!

When one of those discovered missing record fetched with the `DAGResolver` (not received from gRPC) is dispatched to the store: bang!.

Makes more sense?

Reply:
Yep yep, makes sense